### DISCUSSION TITLE
Change partitions on jetson nano devkit

### DISCUSSION THREAD

**Discussion by Achder** [OE4T/meta-tegra#825] [Q&A] [answered]
I'm trying to setup my jetson nano devkit with multiple partitions. 
But I need some guidance on how and where to change the partition configuration. 
All I can find is the mention of the PARTITION_LAYOUT_TEMPLATE in machine/jetson-nano-devkit.conf. But I can't find the actual "flash_l4t_t210_max-spi_sd_p3448.xml" anywhere.
And then there's the final flash.xml.in in the tegraflash which contains the default layout. 
But I don't know where in the yocto project to change it.

**Comment by ichergui** [comment]
Hey @Achder 

Are you using [tegra-demo-distro](https://github.com/OE4T/tegra-demo-distro) ? Are you using `mender` meta layer ? which branch ?
You can find the flash xml file in the build directory `./tmp/work-shared/L4T-tegra-32.x.y/...`
if you are using `mender` meta layer. In this case, mender is overriding the flash xml file to introduce `APP_b` (second root file system partition).

**Reply by Achder** [reply]
I'm creating my own distro. I'm trying to integrate with SWUpdate.

**Comment by madisongh** [comment, marked as answer]
The default layouts are extracted from the L4T BSP, so you'd find them under `$BUILDDIR/tmp/work-shared/L4T-tegra-`_version_ in a build workspace.  If you want to create your own, you would create a recipe for it and add a bbappend for the `tegra-bootfiles` recipe to install it.  The `mender-custom-flash-layout` recipe is one example, and there is another in [my test distro](https://github.com/madisongh/tegra-test-distro/tree/master/layers/meta-testdistro/recipes-bsp/tegra-binaries).

**Reply by Achder** [reply]
Thanks!

**Comment by Achder** [comment]
Just one more question:
For the app partitions you set the size as 

```
<size> APPSIZE </size>
```

How is APPSIZE determined? Same goes for APPFILE.

**Reply by Achder** [reply]
Nevermind, I think I got it: https://github.com/OE4T/meta-tegra/blob/62600c71fe370980c476ee3c487346886a102667/classes/image_types_tegra.bbclass
